src/rabbit_amqqueue.erl

```diff
@@ -397,6 +397,11 @@ not_found_or_absent_dirty(Name) ->
     end.
 
 with(Name, F, E) ->
+    with(Name, F, E, 1000).
+
+with(_Name, _F, E, 0) ->
+    E(infinite_loop_detected);
```

Comment:
It is useful for clients to get a sensible error messages instead of `internal error`.